### PR TITLE
Ensuring ACID Compliance in Database Transactions 

### LINKED ISSUE #732: Ensuring ACID Compliance in Databases
### Brief Summary: 
Describe the what, why, and how of your content idea in 2-5 sentences.

ACID (Atomicity, Consistency, Isolation, Durability) is a set of guiding principles that ensure reliable database transaction processing. A database transaction is any operation performed within a database, such as creating a new record or updating data. Changes made within a database need to be performed with care to ensure the data within does not become corrupted. Applying the ACID properties to each modification of a database is the best way to maintain the database's accuracy and reliability.
### Key Takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?

1. What ACID compliance means  
2. Database transactions 
3. How ACID Works in Practice

### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)
N/A

### PR DESCRIPTION
**Pull request by EKahuha**
this fixes #732

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Review by lalith1403 — APPROVED** [review]
Well written! This discusses ACID properties really well—an overall complete article. Congratulations!

File: articles/ensuring-acid-compliance-in-database-transactions/index.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Another technique to ensure ACID compliance is [shadow paging](https://www.geeksforgeeks.org/shadow-paging-dbms/). This [copy-on-write](https://www.computerhope.com/jargon/c/copy-on-write.htm#) technique avoids in-place page updates and instead allocates a shadow page when a page is modified. You can modify this shadow page liberally without minding consistency constraints because it has no references from other pages on disk. Pages that referred to the original page can only be updated to refer to the new page once the shadow page is ready to become durable. The page is considered automatic because it is activated only when it is ready.
```
Comment (lalith1403) [inline articles/ensuring-acid-compliance-in-database-transactions/index.md:51]: What does liberally mean?   You can modify this shadow page liberally? Check this once.

**Review by lalith1403 — APPROVED** [review]
A minor modification is needed! I have approved the article. Update the explanation, and this should be good to go for the final round of review.

File: articles/ensuring-acid-compliance-in-database-transactions/index.md (hunk 1 above)
Comment (lalith1403) [reply articles/ensuring-acid-compliance-in-database-transactions/index.md:51]: Can you rewrite that sentence?

**Comment by EKahuha** [comment]
change done @lalith1403

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**